backend/EkoFunkcje/EkoFunkcje/Features/Interventions/AddInterventionFunction.cs

```diff
@@ -49,8 +49,8 @@ public AddInterventionFunction(IAddressConverter addressConverter)
             }
             catch (Exception e)
             {
-                log.LogError(e, "error");
-                return new StatusCodeResult(StatusCodes.Status500InternalServerError);
+                log.LogError(e, "Podany adres jest niepoprawny");
```

Comment:
To nas chyba tylko częściowo ratuje.
Rzeczywiście wygląda na to, że nie da się wymusić złej odpowiedzi z Google Maps, kiedy ulica i numer są złe.
Aczkolwiek jest chyba rozwiązanie.

Wydaje mi się, że w AddessConverterze powinnismy sami sprawdzić obiekt odpowiedzi i rzucać odpowiednie wyjątki. 

Nie wiem na ile C# pozwala podczepiać customowe dane do exceptionów (np. Listę błędów), ale jesli pozwala to robił bym coś podobnego (pseudokod - nie znam za dobrze C#)
```
// AddressConverter
public ConvertToGeoAddress() {
   ... pobierz dane z Google Maps ...
   dynamic mapsResult = googleMapsData.results

   var validationResult = validate(mapsResult); // customowy validator
   if (validationResult.notEmpty() {
      var addressException = new BadAddressException(validationResult) // customowy exception
   }
   
    ... zwróc adres jesli validacja przeszła ...
}

public validate(var mapsResult) {
     const validationResults = new List<string>;
     // results: [] oznacza, ze nei znaleziono adresu
     if(isEmpty(mapsResult)) {
         validationResults.add('Podany adres jest niepoprawny');
    }
    if (!hasStreet(mapsResult) {
        validationResults.add('Ulica jest niepoprawna');
    }
    if (!hasStreetNumber(mapsResult) {
        validationResults.add('Numer budynku jest niepoprawny');
    }
}
```

`hasStreet` i `hasSteetNumber` musiałyby przbadać wynik z Google Maps pod względem odpowiednich pól.

Dla poprawnego adresu Geocoding API zwraca coś takiego:
```
{
  "results": [
    {
      "access_points": [],
      "address_components": [
        {
          "long_name": "13",
          "short_name": "13",
          "types": [
            "street_number"
          ]
        },
        {
          "long_name": "Wrocławska",
          "short_name": "Wrocławska",
          "types": [
            "route"
          ]
        },
        {
          "long_name": "Twardogóra",
          "short_name": "Twardogóra",
          "types": [
            "locality",
            "political"
          ]
        },
        {
          "long_name": "Powiat oleśnicki",
          "short_name": "Powiat oleśnicki",
          "types": [
            "administrative_area_level_2",
            "political"
          ]
        },
        {
          "long_name": "Województwo dolnośląskie",
          "short_name": "Województwo dolnośląskie",
          "types": [
            "administrative_area_level_1",
            "political"
          ]
        },
        {
          "long_name": "Poland",
          "short_name": "PL",
          "types": [
            "country",
            "political"
          ]
        },
        {
          "long_name": "56-416",
          "short_name": "56-416",
          "types": [
            "postal_code"
          ]
        }
      ],
      "formatted_address": "Wrocławska 13, 56-416 Twardogóra, Poland",
      "geometry": {
        "location": {
          "lat": 51.3671699,
          "lng": 17.463145
        },
        "location_type": "ROOFTOP",
        "viewport": {
          "northeast": {
            "lat": 51.3685188802915,
            "lng": 17.4644939802915
          },
          "southwest": {
            "lat": 51.3658209197085,
            "lng": 17.4617960197085
          }
        }
      },
      "place_id": "ChIJR9Qynw3_D0cRBgV8bbREmqw",
      "plus_code": {
        "compound_code": "9F87+V7 Twardogora, Poland",
        "global_code": "9F3V9F87+V7"
      },
      "types": [
        "street_address"
      ]
    }
  ]
}
```
Interesujce nas `address_components` dla ulicy i numeru to:
```
 {
          "long_name": "13",
          "short_name": "13",
          "types": [
            "street_number"
          ]
        },
        {
          "long_name": "Wrocławska",
          "short_name": "Wrocławska",
          "types": [
            "route"
          ]
        },
```

W przypadku ich braku oznaczałoby, że Google Maps nie rozpoznał numeru/ulic.
Mozna by też ewentualnie sprawdzać miasto :)